theia-ide/theia
# Fixes to run dart and flutter vscode extensions

#### What it does


Various fixes to run dart and flutter vscode extensions (https://github.com/theia-ide/theia/issues/3999):
- [x] https://github.com/Dart-Code/Dart-Code/issues/1948
- fixes https://github.com/theia-ide/theia/issues/6080

#### How to test


- dart testing
  - install dart and flutter sdk and configure PATH env variable properly
  - install dart and flutter vsix files in plugins folder
    - make sure to use 3.4.0-beta: https://github.com/Dart-Code/Dart-Code/releases/tag/v3.4.0-beta.2, an existing version has known issues, see https://github.com/Dart-Code/Dart-Code/issues/1948
  - start theia
  - use create new project command for flutter
  - open created project
  - use language features
  - debug dart code
  - run dart tests
- [ ] TBD: how to check that view nodes are cleaned up?

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
